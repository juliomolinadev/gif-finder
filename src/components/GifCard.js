import React from "react";
import PropTypes from "prop-types";

const GifCard = ({ gif }) => {
	const { url, title, user, avatar } = gif;

	return (
		<div
			className="card animate__animated animate__fadeInUp animate__slow"
			style={{ width: "18rem" }}
		>
			<img src={url} className="card-img-top" alt={title} />
			<div className="card-body">
				<h5 className="card-title">{title}</h5>

				{user && (
					<div className="d-flex mt-3">
						<img
							className="border border-3 rounded-circle"
							src={avatar}
							alt={user}
							style={{ maxHeight: 30 }}
						/>
						<span className="card-text align-self-center ms-2">{user} </span>
					</div>
				)}

				<a className="btn btn-outline-primary mt-4" href={url} target="_blank" rel="noreferrer">
					Go to resource
				</a>
			</div>
		</div>
	);
};

GifCard.propTypes = {
	gif: PropTypes.object.isRequired
};

export default GifCard;
