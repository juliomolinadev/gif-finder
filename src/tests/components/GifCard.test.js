import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import GifCard from "../../components/GifCard";

describe("Tests in <GifCard />", () => {
	const gifWithUser = {
		url: "https://localhost/any.js",
		title: "a great title",
		user: "julz molina",
		avatar: "https://localhost/img.png"
	};

	const gifWithoutUser = {
		url: "https://localhost/any.js",
		title: "a great title"
	};

	let wrapperWithUser = shallow(<GifCard gif={gifWithUser} />);
	let wrapperWithoutUser = shallow(<GifCard gif={gifWithoutUser} />);

	beforeEach(() => {
		wrapperWithUser = shallow(<GifCard gif={gifWithUser} />);
		wrapperWithoutUser = shallow(<GifCard gif={gifWithoutUser} />);
	});

	test("Component should be displayed correctly", () => {
		expect(wrapperWithUser).toMatchSnapshot();
		expect(wrapperWithoutUser).toMatchSnapshot();
	});

	test("Should have an img whith the url and title in props", () => {
		const img = wrapperWithUser.find("img").at(0);
		expect(img.prop("src")).toBe(gifWithUser.url);
		expect(img.prop("alt")).toBe(gifWithUser.title);
	});

	test("Should have an h5 whith the title in props", () => {
		const title = wrapperWithUser.find("h5");
		expect(title.text().trim()).toBe(gifWithUser.title);
	});

	test("Should have an img whith the avatar and user in props", () => {
		const img = wrapperWithUser.find("img").at(1);
		expect(img.prop("src")).toBe(gifWithUser.avatar);
		expect(img.prop("alt")).toBe(gifWithUser.user);
	});

	test("It shouldn't have 2 img if the props doesn't have a user", () => {
		const img = wrapperWithoutUser.find("img").at(1).exists();
		expect(img).toBe(false);
	});

	test("Should have a link whith the url in props", () => {
		const link = wrapperWithUser.find("a");
		expect(link.prop("href")).toBe(gifWithUser.url);
	});
});
