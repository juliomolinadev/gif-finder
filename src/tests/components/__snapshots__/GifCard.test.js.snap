// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifCard /> Component should be displayed correctly 1`] = `
<div
  className="card"
  style={
    Object {
      "width": "18rem",
    }
  }
>
  <img
    alt="a great title"
    className="card-img-top"
    src="https://localhost/any.js"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      a great title
    </h5>
    <div
      className="d-flex mt-3"
    >
      <img
        alt="julz molina"
        className="border border-3 rounded-circle"
        src="https://localhost/img.png"
        style={
          Object {
            "maxHeight": 30,
          }
        }
      />
      <span
        className="card-text align-self-center ms-2"
      >
        julz molina
         
      </span>
    </div>
    <a
      className="btn btn-primary mt-4"
      href="https://localhost/any.js"
      rel="noreferrer"
      target="_blank"
    >
      Go to resource
    </a>
  </div>
</div>
`;

exports[`Tests in <GifCard /> Component should be displayed correctly 2`] = `
<div
  className="card"
  style={
    Object {
      "width": "18rem",
    }
  }
>
  <img
    alt="a great title"
    className="card-img-top"
    src="https://localhost/any.js"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      a great title
    </h5>
    <a
      className="btn btn-primary mt-4"
      href="https://localhost/any.js"
      rel="noreferrer"
      target="_blank"
    >
      Go to resource
    </a>
  </div>
</div>
`;
